apps/myaccount/src/components/multi-factor-authentication/authenticators/backup-code-authenticator.tsx

```diff
@@ -0,0 +1,490 @@
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { IdentifiableComponentInterface } from "@wso2is/core/models";
+import { CommonUtils } from "@wso2is/core/utils";
+import { ContentLoader, Heading, LinkButton } from "@wso2is/react-components";
+import React, { MouseEvent, PropsWithChildren, useEffect, useState } from "react";
+import { useTranslation } from "react-i18next";
+import { useSelector } from "react-redux";
+import {
+    Button,
+    Container,
+    Grid,
+    GridColumn,
+    Icon,
+    Label,
+    List,
+    Message,
+    Modal,
+    Popup,
+    Segment    
+} from "semantic-ui-react";
+import {
+    generateBackupCodes,
+    getRemainingBackupCodesCount
+} from "../../../api";
+import {
+    AlertInterface,
+    AlertLevels,
+    AuthStateInterface,
+    BackupCodeInterface,
+    BackupCodesCountInterface
+} from "../../../models";
+import { AppState } from "../../../store";
+
+/**
+ * Property types for the backup code component.
+ * Also see {@link BackupCodeAuthenticator.defaultProps}
+ */
+interface BackupCodeProps extends IdentifiableComponentInterface {
+    onAlertFired: (alert: AlertInterface) => void;
+    initBackupCodeFlow: boolean;
+    onBackupFlowCompleted: () => void;
+}
+
+export const BackupCodeAuthenticator : React.FunctionComponent<BackupCodeProps> = (
+    props: PropsWithChildren<BackupCodeProps>
+): React.ReactElement => {
+
+    const { 
+        onAlertFired,
+        initBackupCodeFlow,
+        onBackupFlowCompleted,
+        ["data-componentid"]: componentid 
+    } = props;
+
+    const { t } = useTranslation();
+
+    const translateKey: string = "myAccount:components.mfa.backupCode.";
+
+    const profileDetails: AuthStateInterface = useSelector(
+        (state: AppState) => state.authenticationInformation
+    );
+
+    const [ isLoading, setIsLoading ] = useState<boolean>(false);
+    const [ isModalOpen, setIsModalOpen ] = useState<boolean>(false);
+    const [ backupCodes, setBackupCodes ] = useState<Array<string>>([]);
+    const [ remainingBackupCodes, setRemainingBackupCodes ] = useState<number>(0);
+    const [ isCodesCopied, setIsCodesCopied ] = useState<boolean>(false);
+    const [ isCodesDownloaded, setIsCodesDownloaded ] = useState<boolean>(false);
+    const [ isConfirmRegenerationModalOpen, setIsConfirmRegenerationModalOpen ] = useState<boolean>(false);
+    const [ isWarnRemaingBackupCodes, setIsWarnRemaingBackupCodes ] = useState<boolean>(false);
+
+    const minBackupCodesLimit: number = 4;
+    
+    /**
+     * Starts backup code configuration flow
+     */
+    useEffect(() => {         
+        if (initBackupCodeFlow) {
+            setIsLoading(true);
+            setIsModalOpen(true);
+            initBackupCodes();
+        }
+    }, [ initBackupCodeFlow ]);
+
+    /**
+     * Fetch remaining backup codes count
+     */
+    useEffect(() => {
```

Comment:
Move this to the top for better readability.

Reply:
Fixed with 0250f21d2f112990b7cc67530b74f52609d0c7e8